### DISCUSSION TITLE
How to trigger onError for cached data?

### DISCUSSION THREAD

**Discussion by winlx** [TanStack/query#2744] [Q&A]
When data is taken from the cache, the global onError is not triggered and popup error is not displayed. It turns out that onError is useless if data caching ?

**Comment by TkDodo** [comment]
Can you be a bit more precise please, maybe with a codesandbox example of the issues you are facing

**Reply by winlx** [reply]
onError is triggered only at the first request for a specific endpoint. Subsequent requests to the same endpoint are returned from the cache and onError is not triggered.

**Reply by TkDodo** [reply]
I don’t think that’s true, at least not to my knowledge. Erroneous queries are always considered stale, so with the next smart refetch point, it should go back to loading state and refetch.

**Reply by winlx** [reply]
Maybe there is some example showing how it works?

**Reply by TkDodo** [reply]
going back to my initial request, please come up with a codesandbox example that shows your issue, then we can look at it and see what's going on.

**Reply by winlx** [reply]
OMG! It was a bug: core: make sure erroneous queries refetch when query keys change (#2686) (1beedf9)
